algolia/transport/transport.go

```diff
@@ -302,10 +304,14 @@ func buildRequest(
 }
 
 func unmarshalTo(r io.ReadCloser, v interface{}) error {
-	err := json.NewDecoder(r).Decode(&v)
+	body, err := ioutil.ReadAll(r)
 	errClose := r.Close()
 	if err != nil {
-		return fmt.Errorf("cannot deserialize response's body: %v", err)
+		return fmt.Errorf("cannot read body: %v", err)
+	}
+	err = json.NewDecoder(bytes.NewReader(body)).Decode(&v)
```

Comment:
```suggestion
	err = json.Unmarshal(body, &v)
```

We already have the data in `[]byte` we don't have to create a new reader for that. We can call unmarshal.